package.json

```diff
@@ -58,6 +58,7 @@
     "eth-sig-util": "^3.0.0",
     "ethereumjs-util": "^7.0.10",
     "ethereumjs-wallet": "^1.0.1",
+    "ethers": "^5.4.1",
```

Comment:
I elected to introduce ethers to this repo because I figure we want to migrate away from web3 as our ethereum querying package.

Reply:
We do, but we don't yet have a provider that works with ethers. It expects the provider to be EIP-1192 compliant, and the one we use internally in the extension isn't compliant with any spec :sweat_smile: It only has one method, `sendAsync`.

It's not clear exactly what works and what doesn't when it's given a provider like the one we use. The only thing I know doesn't work is that the provider breaks when the network changes. We had to destroy it and reconstruct it upon network change last time we introduced it in swaps.

Reply:
Ok hmmm. So we were using this same passed in provider with ethers in the extension-based version of this functionality. Its initialized [here](https://github.com/MetaMask/metamask-extension/blob/a294f02b1d46a16cc443b627f2b1e492809177b1/app/scripts/controllers/preferences.js#L83). From what I can tell it works in the same way. Though, as you point out I do need to recreate a new instance on network change. Does that address concerns? Given that it mimics existing behavior?

Reply:
Perhaps so. I'm not even sure our usage of ethers in the extension is correct though, because we don't know what other 1193 events it depends upon. Just that there are no known issues.

I can dig in to try to better understand that I suppose.

Reply:
@Gudahtt is it okay to make that enhancement in a separate PR? Since this is not regressing the existing functionality (however limited it may be)?

Reply:
Sorry I didn't mean to suggest any enhancements. I was talking about the usage you're introducing here in this PR.

Reply:
Ah yes so I guess I'm thinking of "the usage [i'm] introducing here" as replicating the existing usage in the preferences controller in the extension, making the work on getting together an EIP-1193 compliant provider an enhancement in my mind. But I guess your thinking of it differently because this is a new repo where I'm introducing a wrong usage?

Reply:
Oh, sheesh, my bad! I didn't notice that the exact same call was being made in the preferences controller already. You're absolutely right, this makes sense.